File: frontend/src/apis/memory.ts
import request from './request'

// 记忆类型定义
export interface Memory {
  memory_id: string
  content: string
  memory_type: string
  source_dialog_id: string
  created_at: number | null
  importance: number
}

export interface MemoryListResponse {
  total: number
  items: Memory[]
}

export interface MemoryCreateRequest {
  content: string
  memory_type: string
  importance?: number
}

export interface MemoryUpdateRequest {
  content?: string
  memory_type?: string
  importance?: number
}

// 获取记忆列表（支持分页和搜索）
export const getMemoryList = (params?: {
  page?: number
  page_size?: number
  keyword?: string
  memory_type?: string
}) => {
  return request.get<MemoryListResponse>('/memory/', { params })
}

// 手动添加记忆
export const createMemory = (data: MemoryCreateRequest) => {
  return request.post<Memory>('/memory/', data)
}

// 更新记忆
export const updateMemory = (memoryId: string, data: MemoryUpdateRequest) => {
  return request.put<Memory>(`/memory/${memoryId}`, data)
}

// 删除记忆
export const deleteMemory = (memoryId: string) => {
  return request.delete(`/memory/${memoryId}`)
}

File: backend/app/schemas/memory.py
"""长期记忆相关 Schema"""
from pydantic import BaseModel
from typing import Optional, List


class MemoryResponse(BaseModel):
    """记忆响应"""
    memory_id: str
    content: str
    memory_type: str
    source_dialog_id: Optional[str] = ""
    created_at: Optional[int] = None
    importance: int = 3


class MemoryListResponse(BaseModel):
    """记忆列表响应"""
    total: int
    items: List[MemoryResponse]


class MemoryCreateRequest(BaseModel):
    """手动添加记忆请求"""
    content: str
    memory_type: str = "fact"
    importance: int = 3


class MemoryUpdateRequest(BaseModel):
    """更新记忆请求"""
    content: Optional[str] = None
    memory_type: Optional[str] = None
    importance: Optional[int] = None
